### ISSUE TITLE
CI:  k8s-1.7.K8sKafkaPolicyTest Kafka Policy Tests KafkaPolicies: failed to produce empire-hq on topic empire-announce 

### ISSUE THREAD

**Issue by raybejjani** [cilium/cilium#5164]
I think the root cause here is that both nodes seem very overloaded.
```
cmd: "kubectl exec -n kube-system cilium-3r78t -- uptime" exitCode: 0 duration: 1m14.024303377s stdout:
 23:11:31 up 54 min,  0 users,  load average: 49.93, 24.52, 11.62
```
the uptime command itself took a minute to complete, and the load averages are also high.

It's possible that this is a follow-on to https://github.com/cilium/cilium/issues/4951 but I'm not sure.

[ac1b5e7e_K8sKafkaPolicyTest_Kafka_Policy_Tests_KafkaPolicies.zip](https://github.com/cilium/cilium/files/2277471/ac1b5e7e_K8sKafkaPolicyTest_Kafka_Policy_Tests_KafkaPolicies.zip)

Stacktrace
```
/home/jenkins/workspace/cilium-ginkgo_cilium_master-QVKDPVJ2A6E272IRUX7X6DCB5FBLCQDKRTOMHPZS7ZECX3RM5C3A/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:376
Failed to produce to empire-hq on topic empire-announce
Expected
    <*errors.errorString | 0xc4204c6150>: {
        s: "ExecKafkaPodCmd: command 'kubectl exec -n default empire-hq-2917507631-kq0pg sh -- -c \"echo 'Happy 40th Birthday to General Tagge' | ./kafka-produce.sh --topic empire-announce\"' failed 'Exitcode: 0 \nStdout:\n \t >>\nStderr:\n \t [2018-08-09 23:11:27,574] ERROR Error when sending message to topic empire-announce with key: null, value: 36 bytes with error: (org.apache.kafka.clients.producer.internals.ErrorLoggingCallback)\n\t org.apache.kafka.common.errors.TimeoutException: Expiring 1 record(s) for empire-announce-0: 3915 ms has passed since batch creation plus linger time\n\t \n'",
    }
to be nil
/home/jenkins/workspace/cilium-ginkgo_cilium_master-QVKDPVJ2A6E272IRUX7X6DCB5FBLCQDKRTOMHPZS7ZECX3RM5C3A/src/github.com/cilium/cilium/test/k8sT/KafkaPolicies.go:171
Standard Output
Number of "context deadline exceeded" in logs: 0
Number of "level=error" in logs: 1
Number of "level=warning" in logs: 0
Number of "Cilium API handler panicked" in logs: 0
Cilium pods: [cilium-3r78t cilium-90l0g]
Netpols loaded: 
CiliumNetworkPolicies loaded: default::cnp-second-namespace 
Endpoint Policy Enforcement:
     kafka-broker-2873197211-lqghj          =>   none
     zookeeper-1032194181-0zlvp             =>   none
     kube-dns-2552250787-4pr1l              =>   none
     prometheus-core-3506541099-61ssr       =>   none
     empire-backup-2986355077-jstj9         =>   none
     empire-hq-2917507631-kq0pg             =>   none
     empire-outpost-8888-3627733903-nv32t   =>   none
     empire-outpost-9999-4143713519-fmp5t   =>   none
Cilium agent "cilium-3r78t": Status: Ok  Health: Warning Nodes "k8s2" ContinerRuntime: Ok Kubernetes: Ok KVstore: Ok Controllers: Total 39 Failed 0
Cilium agent "cilium-90l0g": Status: Ok  Health: Ok Nodes "k8s2 k8s1" ContinerRuntime: Ok Kubernetes: Ok KVstore: Ok Controllers: Total 37 Failed 0

Standard Error
STEP: Wait for Kafka broker to be up
STEP: Creating new kafka topic empire-announce
STEP: Creating new kafka topic deathstar-plans
STEP: Waiting for DNS to resolve within pods for kafka-service
STEP: Testing basic Kafka Produce and Consume
=== Test Finished at 2018-08-09T23:11:30Z====
===================== TEST FAILED =====================
cmd: kubectl get pods -o wide --all-namespaces
NAMESPACE     NAME                                   READY     STATUS    RESTARTS   AGE       IP              NODE
default       empire-backup-2986355077-jstj9         1/1       Running   0          3m        10.10.0.135     k8s1
default       empire-hq-2917507631-kq0pg             1/1       Running   0          3m        10.10.0.55      k8s1
default       empire-outpost-8888-3627733903-nv32t   1/1       Running   0          3m        10.10.1.91      k8s2
default       empire-outpost-9999-4143713519-fmp5t   1/1       Running   0          3m        10.10.1.51      k8s2
default       kafka-broker-2873197211-lqghj          1/1       Running   0          3m        10.10.0.19      k8s1
default       zookeeper-1032194181-0zlvp             1/1       Running   0          3m        10.10.0.53      k8s1
kube-system   cilium-3r78t                           1/1       Running   0          32m       192.168.36.11   k8s1
kube-system   cilium-90l0g                           1/1       Running   0          32m       192.168.36.12   k8s2
kube-system   etcd-k8s1                              1/1       Running   0          49m       192.168.36.11   k8s1
kube-system   kube-apiserver-k8s1                    1/1       Running   0          49m       192.168.36.11   k8s1
kube-system   kube-controller-manager-k8s1           1/1       Running   1          49m       192.168.36.11   k8s1
kube-system   kube-dns-2552250787-4pr1l              3/3       Running   3          33m       10.10.1.141     k8s2
kube-system   kube-proxy-8sb7b                       1/1       Running   0          43m       192.168.36.12   k8s2
kube-system   kube-proxy-z9j0s                       1/1       Running   0          50m       192.168.36.11   k8s1
kube-system   kube-scheduler-k8s1                    1/1       Running   0          50m       192.168.36.11   k8s1
prometheus    prometheus-core-3506541099-61ssr       1/1       Running   0          43m       10.10.1.228     k8s2

cmd: kubectl exec -n kube-system cilium-3r78t -- cilium service list
ID   Frontend              Backend                   
1    10.96.0.1:443         1 => 192.168.36.11:6443   
2    10.107.255.255:9090   1 => 10.10.1.228:9090     
4    10.96.0.10:53         1 => 10.10.1.141:53       
20   10.100.60.241:2181    1 => 10.10.0.53:2181      

cmd: kubectl exec -n kube-system cilium-3r78t -- cilium endpoint list
ENDPOINT   POLICY (ingress)   POLICY (egress)   IDENTITY   LABELS (source:key[=value])                       IPv6                 IPv4          STATUS   
           ENFORCEMENT        ENFORCEMENT                                                                                                       
5005       Disabled           Disabled          9695       k8s:app=zook                                      f00d::a0a:0:0:138d   10.10.0.53    ready   
                                                           k8s:io.cilium.k8s.policy.serviceaccount=default                                              
                                                           k8s:io.kubernetes.pod.namespace=default                                                      
                                                           k8s:zgroup=kafkaTestApp                                                                      
5924       Disabled           Disabled          43153      k8s:app=empire-backup                             f00d::a0a:0:0:1724   10.10.0.135   ready   
                                                           k8s:io.cilium.k8s.policy.serviceaccount=default                                              
                                                           k8s:io.kubernetes.pod.namespace=default                                                      
                                                           k8s:zgroup=kafkaTestApp                                                                      
25611      Disabled           Disabled          19636      k8s:app=empire-hq                                 f00d::a0a:0:0:640b   10.10.0.55    ready   
                                                           k8s:io.cilium.k8s.policy.serviceaccount=default                                              
                                                           k8s:io.kubernetes.pod.namespace=default                                                      
                                                           k8s:zgroup=kafkaTestApp                                                                      
28993      Disabled           Disabled          65125      k8s:app=kafka                                     f00d::a0a:0:0:7141   10.10.0.19    ready   
                                                           k8s:io.cilium.k8s.policy.serviceaccount=default                                              
                                                           k8s:io.kubernetes.pod.namespace=default                                                      
                                                           k8s:zgroup=kafkaTestApp                                                                      

cmd: kubectl exec -n kube-system cilium-90l0g -- cilium service list
ID   Frontend              Backend                   
1    10.96.0.1:443         1 => 192.168.36.11:6443   
2    10.107.255.255:9090   1 => 10.10.1.228:9090     
5    10.96.0.10:53         1 => 10.10.1.141:53       
20   10.100.60.241:2181    1 => 10.10.0.53:2181      

cmd: kubectl exec -n kube-system cilium-90l0g -- cilium endpoint list
ENDPOINT   POLICY (ingress)   POLICY (egress)   IDENTITY   LABELS (source:key[=value])                              IPv6                   IPv4          STATUS   
           ENFORCEMENT        ENFORCEMENT                                                                                                                
3424       Disabled           Disabled          4374       k8s:app=prometheus                                       f00d::a0a:100:0:d60    10.10.1.228   ready   
                                                           k8s:component=core                                                                                    
                                                           k8s:io.cilium.k8s.policy.serviceaccount=prometheus-k8s                                                
                                                           k8s:io.kubernetes.pod.namespace=prometheus                                                            
4124       Disabled           Disabled          1815       k8s:app=empire-outpost                                   f00d::a0a:100:0:101c   10.10.1.91    ready   
                                                           k8s:io.cilium.k8s.policy.serviceaccount=default                                                       
                                                           k8s:io.kubernetes.pod.namespace=default                                                               
                                                           k8s:outpostid=8888                                                                                    
                                                           k8s:zgroup=kafkaTestApp                                                                               
25367      Disabled           Disabled          4          reserved:health                                          f00d::a0a:100:0:6317   10.10.1.113   ready   
43545      Disabled           Disabled          63177      k8s:io.cilium.k8s.policy.serviceaccount=kube-dns         f00d::a0a:100:0:aa19   10.10.1.141   ready   
                                                           k8s:io.kubernetes.pod.namespace=kube-system                                                           
                                                           k8s:k8s-app=kube-dns                                                                                  
54859      Disabled           Disabled          36972      k8s:app=empire-outpost                                   f00d::a0a:100:0:d64b   10.10.1.51    ready   
                                                           k8s:io.cilium.k8s.policy.serviceaccount=default                                                       
                                                           k8s:io.kubernetes.pod.namespace=default                                                               
                                                           k8s:outpostid=9999                                                                                    
                                                           k8s:zgroup=kafkaTestApp                                                                               

===================== Existing AfterFailed =====================

```

**Comment by eloycoto** [comment]
Just a quick note: `Warning Nodes "k8s2"`

**Comment by tgraf** [comment]
Fixed via #5156
